### PR TITLE
Merge stable branch into main for 4.0 release prep, add 4.0 changelog entry

### PR DESCRIPTION
**Pull request by munkm**
This PR merges the stable branch into main in preparation for the 4.0 release. This means main should now have the 3x series tag in the history. I will issue a separate PR after the release to get the 4.0 tag on main. 

This PR also includes the summary documentation for the changelog for 4.0.

### CODE REVIEW

File: doc/source/reference/changelog.rst [hunk 1]
```diff
@@ -12,6 +12,250 @@ The `CREDITS file <https://github.com/yt-project/yt/blob/main/CREDITS>`_
...
+   #\ `2869 <https://github.com/yt-project/yt/pull/2869>`__ and PR
+   #\ `3094 <https://github.com/yt-project/yt/pull/3094>`__)
+-  ``list_annotations`` method for plots (PR
+   #\ `2562 <https://github.com/yt-project/yt/pull/2562>`__)
+
+API improvements
+^^^^^^^^^^^^^^^^
+
+-  ``yt.load`` with support for ``os.Path``-like objects, improved UX
+   and refactored into its own module (PR
```
Comment (neutrinoceros) [inline doc/source/reference/changelog.rst:51]:
```suggestion
   and moved a new ``yt.loaders`` module, along with sibling functions (PR
```

**Review by neutrinoceros — COMMENTED** [review]
Looks good, thanks for cleaning this up ! just a few minor corrections here

**Comment by neutrinoceros** [comment]
sorry I manually relaunched CI jobs by accident

**Comment by munkm** [comment]
pre-commit.ci autofix

**Review by neutrinoceros — APPROVED** [review]

**Pull request merged**